OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -62,6 +62,9 @@ Joint::Joint(const std::string &name, const std::string& parentName,
                                       const std::string& childName,
                                       bool reverse) : Joint()
 {
+    OPENSIM_THROW_IF( name.empty(), ComponentHasNoName,
+                      getClassName());
+
```

Comment:
I think the elegant way to handle this is that the constructor has a signature

```
Joint::Joint(const ComponentPathName& name, const ComponentPathName& parent, const ComponentPathName& child)
```

and the constructor to `ComponentPathName` does all the validation.